gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/plugin/Plugin.kt

```diff
@@ -337,21 +337,21 @@ private fun configureCompileTasks(
     javaCompile: JavaCompile,
     kotlinCompile: KotlinCompile<*>?
 ) {
-    // The predicate to filter out files from `build/generated-proto` directory.
-    val notInSourceDir: (File) -> Boolean = { file -> !file.residesIn(sourceDir) }
-
-    // Re-set sources for `JavaCompile` assuming filtering.
-    javaCompile.source = javaCompile.source.filter(notInSourceDir).asFileTree
-
-    // Do the same for `KotlinCompile`, if it's present in the project.
-    @Suppress("IfThenToSafeAccess") // Looks more readable.
-    if (kotlinCompile is KotlinCompileTool) {
-        val filteredKotlin = kotlinCompile.sources.filter(notInSourceDir).toSet()
-        with(kotlinCompile.sources as ConfigurableFileCollection) {
-            setFrom(filteredKotlin)
-        }
-    } else if (kotlinCompile != null) {
-        kotlinCompile.project.logger.warn(
+    val pathToExclude = sourceDir.asFile.canonicalPath
```

Comment:
Please document this. Not knowing the context makes it looks really weird: “Are we excluding source directory from compilation? What?..”